core/src/main/scala/com/mesosphere/usi/core/SchedulerLogic.scala

```diff
@@ -0,0 +1,251 @@
+package com.mesosphere.usi.core
+
+import java.time.Instant
+
+import com.mesosphere.usi.core.models._
+import com.mesosphere.usi.models._
+
+object SchedulerLogic {
+  private def assertValidTransition(oldSpec: PodSpec, newSpec: PodSpec): Unit = {
+    if ((oldSpec.goal == Goal.Terminal) && (newSpec.goal == Goal.Running))
+      throw new IllegalStateException(s"Illegal state transition Terminal to Running for podId ${oldSpec.id}")
```

Comment:
How are exceptions handled and where? Should this bubble up to the orchestration layer?

Reply:
Ideally we'd crash.

Also, this is a mock implementation.